spec/lib/http/uri_spec.rb

```diff
@@ -29,4 +29,18 @@
       expect(http_uri.to_s).to eq("http://example.com")
     end
   end
+
+  describe ".encode_query" do
```

Comment:
It worth to add spec that covers problem explained in the issue:

``` ruby
describe "#request" do
  it "correctly encodes new line chars in query params" do
    str = "bar\nbaz"
    res = HTTP.get("http://httpbin.org/get", :params => { :foo => str })
    expect(res.parse(:json).dig("args", "foo")).to eq str
  end
end

```

Reply:
I think this minor string generation bug is covered sufficiently by a unit test and isn't worth a high-level integration test like that.

(I'm assuming the project has some integration tests like you described, to confirm basic functionality.)